### PR TITLE
Bump mysql image used for testing for improved arm64 compatibility

### PR DESCRIPTION
**Pull request by ivoanjo**
**What does this PR do?**:

Upgrade the mysql image used for testing in CI to a more modern version.

**Motivation**:

According to @TonyCTHsu this version works well on arm64, and it still seems to work fine on all Ruby versions in CI, so this is more of a "why not" change.

**Additional Notes**: 

This is a follow-up to #2171.

**How to test the change?**:

* Check that our mysql integration tests are still passing.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -581,7 +581,7 @@ services:
     ports:
       - "${TEST_MONGODB_PORT}:27017"
   mysql:
-    image: mysql:5.6
+    image: mysql:8
     platform: linux/x86_64
```
Comment (TonyCTHsu) [inline docker-compose.yml:585]: I think we should remove the `platform` attribute to allow docker to pick up different architecture base on host machine.

Comment (ivoanjo) [reply docker-compose.yml:585]: Uh, right. Oopsie! Will do!

Comment (ivoanjo) [reply docker-compose.yml]: Fixed in 8a12f865083f83aba3d39e822232902bbebee8d8

**Comment by TonyCTHsu** [comment]
I can confirm the image `02662e079c08` I pulled are green with `bundle exec appraisal ruby-3.0.3-contrib rake spec:mysql2`

**Review by TonyCTHsu — APPROVED** [review]

**Pull request merged**